[2025-04-08T03:39:33.590+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-08T03:39:33.881+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: scrape_amazon_dag.process_Paint___Paint_Supplies_7 manual__2025-04-08T03:39:12.456722+00:00 [queued]>
[2025-04-08T03:39:33.915+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: scrape_amazon_dag.process_Paint___Paint_Supplies_7 manual__2025-04-08T03:39:12.456722+00:00 [queued]>
[2025-04-08T03:39:33.916+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 2
[2025-04-08T03:39:39.652+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): process_Paint___Paint_Supplies_7> on 2025-04-08 03:39:12.456722+00:00
[2025-04-08T03:39:41.034+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=5389) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-04-08T03:39:41.040+0000] {standard_task_runner.py:72} INFO - Started process 5442 to run task
[2025-04-08T03:39:41.052+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'scrape_amazon_dag', 'process_Paint___Paint_Supplies_7', 'manual__2025-04-08T03:39:12.456722+00:00', '--job-id', '150', '--raw', '--subdir', 'DAGS_FOLDER/test_crawl_list_product.py', '--cfg-path', '/tmp/tmpdvguk80e']
[2025-04-08T03:39:41.054+0000] {standard_task_runner.py:105} INFO - Job 150: Subtask process_Paint___Paint_Supplies_7
[2025-04-08T03:39:41.410+0000] {task_command.py:467} INFO - Running <TaskInstance: scrape_amazon_dag.process_Paint___Paint_Supplies_7 manual__2025-04-08T03:39:12.456722+00:00 [running]> on host 13b4251b7529
[2025-04-08T03:39:41.900+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='scrape_amazon_dag' AIRFLOW_CTX_TASK_ID='process_Paint___Paint_Supplies_7' AIRFLOW_CTX_EXECUTION_DATE='2025-04-08T03:39:12.456722+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-08T03:39:12.456722+00:00'
[2025-04-08T03:39:41.906+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-08T03:39:41.906+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-08T03:39:41.907+0000] {logging_mixin.py:190} INFO - Current task name:process_Paint___Paint_Supplies_7 state:running start_date:2025-04-08 03:39:33.884935+00:00
[2025-04-08T03:39:41.908+0000] {logging_mixin.py:190} INFO - Dag name:scrape_amazon_dag and current dag run status:running
[2025-04-08T03:39:41.909+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-04-08T03:39:41.911+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: scrape_department() missing 1 required positional argument: 'user_agent'
[2025-04-08T03:39:41.946+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-04-08T03:39:41.947+0000] {logging_mixin.py:190} INFO - Task start:2025-04-08 03:39:33.884935+00:00 end:2025-04-08 03:39:41.945167+00:00 duration:8.060232
[2025-04-08T03:39:41.948+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): process_Paint___Paint_Supplies_7> dag:<DAG: scrape_amazon_dag> dagrun:<DagRun scrape_amazon_dag @ 2025-04-08 03:39:12.456722+00:00: manual__2025-04-08T03:39:12.456722+00:00, state:running, queued_at: 2025-04-08 03:39:12.494777+00:00. externally triggered: True>
[2025-04-08T03:39:41.949+0000] {logging_mixin.py:190} INFO - Failure caused by scrape_department() missing 1 required positional argument: 'user_agent'
[2025-04-08T03:39:41.950+0000] {taskinstance.py:1226} INFO - Marking task as UP_FOR_RETRY. dag_id=scrape_amazon_dag, task_id=process_Paint___Paint_Supplies_7, run_id=manual__2025-04-08T03:39:12.456722+00:00, execution_date=20250408T033912, start_date=20250408T033933, end_date=20250408T033941
[2025-04-08T03:39:42.110+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-04-08T03:39:42.116+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 150 for task process_Paint___Paint_Supplies_7 (scrape_department() missing 1 required positional argument: 'user_agent'; 5442)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: scrape_department() missing 1 required positional argument: 'user_agent'
[2025-04-08T03:39:42.239+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-04-08T03:39:42.260+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
